src/server/webserver.ts

```diff
@@ -169,13 +169,13 @@ export class WebServer {
       this.relaywss.on('connection', async (ws, req) => {
         try {
           const base = `${this.config.https ? 'https' : 'http'}://${this.config.common_name}:${this.config.web_port}/`
-          const reqQueryUrl = new URL(req.url, base)
-          req.query = {
-            host: reqQueryUrl.searchParams.get('host'),
-            port: reqQueryUrl.searchParams.get('port'),
-            p: reqQueryUrl.searchParams.get('p'),
-            tls: reqQueryUrl.searchParams.get('tls'),
-            tls1only: reqQueryUrl.searchParams.get('tls1only')
+          const reqQueryURL = new URL(req.url, base)
+          const params: queryParams = {
+            host: reqQueryURL.searchParams.get('host'),
+            port: Number(reqQueryURL.searchParams.get('port')),
```

Comment:
where does Number come from? why not parseInt()?

Reply:
Number function also can be used in TP to convert string.